### DISCUSSION TITLE
Farfield plot for array in 2D and 3D for all parametric sweeps

### DISCUSSION THREAD

**Discussion by Abrinia** [ansys/pyaedt#4866] [Q&A] [answered]
Hello all,

Is there a way to plot all 2D cut plots for all theta and given phi and 3D plot for each source defined in design with all parametric variations in different plots?

Thanks.

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi, yes it is possible. Please take a look at this example:

https://aedt.docs.pyansys.com/version/stable/examples/02-HFSS/Array.html#load-far-field-data

You can get the ffdata, and you will have all far field data, you just need to manipulate the data as you prefer.

**Reply by Samuelopez-ansys** [reply]
@Abrinia

**Reply by Abrinia** [reply]
Hello Samuel,
Thanks for reply. But I am asking if I have multiple sources and parametric sweeps, usually it appears in source context: Port1:1 or lets say port 2:1 for array. 
hfss.post.create_report(expressions="dB(GainTotal)", setup_sweep_name=f"{setup_name[0]} : {sweep_name[1]}", variations = vars,  primary_sweep_variable = "Theta", context = "E_plane", plot_type="Rectangular Plot", report_category = "Far Fields", plot_name="Gain Plot")

Here, how can I define different sources to create a FF plot. E_plane is farfield sphere defined
